CLAUDE.md

```diff
@@ -1,575 +1,779 @@
-# Bitwarden MCP Server
+# Bitwarden MCP Server - Developer Guide
 
-Security-critical TypeScript MCP server providing AI models with access to Bitwarden through two complementary interfaces:
+This document provides comprehensive guidelines for developing and maintaining the Bitwarden MCP Server.
 
-- **Vault Management**: Personal vault operations via Bitwarden CLI
-- **Organization Administration**: Enterprise admin functions via Bitwarden Public API
+## Architecture Overview
 
-**Stack:** TypeScript ES modules, MCP SDK, Zod validation, Node.js 22, HTTP client
+### Dual Interface Pattern
+
+The MCP server exposes two distinct operational interfaces:
+
+**1. CLI Interface (Vault Management and CLI Tools)**
+
+- Wraps Bitwarden CLI (`bw`) commands for personal vault operations
+- Requires `BW_SESSION` environment variable
+- Executes shell commands with security hardening
+- Returns plain text or JSON responses from CLI
+
+**2. API Interface (Organization Administration)**
 
-## 🔒 Dual Security Architecture (MANDATORY)
+- Makes authenticated HTTP requests to Bitwarden Public API
+- Requires OAuth2 client credentials (`BW_CLIENT_ID`, `BW_CLIENT_SECRET`)
+- Uses RESTful JSON communication
+- Returns structured JSON responses
+
+### Request Flow
+
+```
+AI Client (Claude Desktop)
+    ↓
+MCP Protocol (stdio transport)
+    ↓
+index.ts (tool routing)
+    ↓
+┌─────────────────────┬──────────────────────┐
+│   CLI Handler       │   API Handler        │
+│   ↓                 │   ↓                  │
+│   Security Layer    │   OAuth2 Client      │
+│   ↓                 │   ↓                  │
+│   Bitwarden CLI     │   Bitwarden API      │
+└─────────────────────┴──────────────────────┘
+```
 
-### CLI Tools Security Pipeline
+## Code Organization
 
-For vault management operations, every tool MUST follow this pattern - NO EXCEPTIONS:
+### Directory Structure
+
+```
+src/
+├── index.ts              # MCP server setup and tool routing
+├── handlers/             # Business logic implementations
+│   ├── cli.ts            # CLI command handlers (30 tools)
```

Comment:
🎨 I recommend taking these numbers out all throughout here. No need to maintain that.